catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -387,6 +406,41 @@ public CatalogFrameworkImpl(List<CatalogProvider> catalogProviders, BundleContex
         }
     }
 
+    /**
+     * Instantiates a new CatalogFrameworkImpl
+     * @param frameworkProperties - collection of properties to be set on the CatalogFramework instance
+     */
+    public CatalogFrameworkImpl(FrameworkProperties frameworkProperties) {
+        this.context = frameworkProperties.getBundleContext();
+        this.catalogProviders = frameworkProperties.getCatalogProviders();
+        if (LOGGER.isDebugEnabled()) {
+            if (this.catalogProviders != null) {
+                LOGGER.debug("catalog providers list size = " + this.catalogProviders.size());
+            } else {
+                LOGGER.debug("catalog providers list is NULL");
+            }
+        }
+
+        this.preIngest = frameworkProperties.getPreIngest();
+        this.postIngest = frameworkProperties.getPostIngest();
+        this.preQuery = frameworkProperties.getPreQuery();
+        this.postQuery = frameworkProperties.getPostQuery();
+        this.preResource = frameworkProperties.getPreResource();
+        this.postResource = frameworkProperties.getPostResource();
+        this.connectedSources = frameworkProperties.getConnectedSources();
+        this.federatedSources = frameworkProperties.getFederatedSources();
+        this.resourceReaders = frameworkProperties.getResourceReaders();
+        this.defaultFederationStrategy = frameworkProperties.getFederationStrategy();
+        this.queryResponsePostProcessor = frameworkProperties.getQueryResponsePostProcessor();
+        this.poller = frameworkProperties.getSourcePoller();
+        this.productCache = frameworkProperties.getResourceCache();
+        this.retrieveStatusEventPublisher = frameworkProperties.getDownloadsStatusEventPublisher();
+        this.reliableResourceDownloadManager = frameworkProperties.getReliableResourceDownloadManager();
+        synchronized (this) {
```

Comment:
I'm not clear on that either, that is how it was done so I just kept it. I'll remove it though. It looks like it is not needed to me.